### PR TITLE
Add lint to frontend

### PR DESCRIPTION
**Pull request by binkpitch**
This PR adds lint to frontend using eslint and prettier (as well as husky and lint-stage).

Ref: #120 

------

Not sure if I should fix lint in this pr.

### CODE REVIEW

**Review by lelouchB — APPROVED** [review]

File: frontend/package.json [hunk 1]
```diff
@@ -1,52 +1,77 @@
...
+  "scripts": {
+    "docusaurus": "docusaurus",
+    "start": "docusaurus start",
+    "build": "docusaurus build",
+    "swizzle": "docusaurus swizzle",
+    "deploy": "docusaurus deploy",
+    "serve": "docusaurus serve",
+    "lint": "eslint .",
+    "lint:fix": "eslint --fix --ext .js,.jsx .",
+    "format": "prettier --write './**/*.{js,jsx,css,md,json}'"
```
Comment (lelouchB) [inline frontend/package.json:13]: Hey this command is not working properly 
Can you fix this
`npm run format`

**Review by lelouchB — CHANGES_REQUESTED** [review]

File: frontend/package.json (hunk 1 above)
Comment (binkpitch) [reply frontend/package.json:13]: Can you explain more on what isn't working properly

Comment (lelouchB) [reply frontend/package.json:13]: This command 
` npm run format`

Does not run on  my machine

Does it run on yours?

Comment (binkpitch) [reply frontend/package.json:13]: It runs correctly on my machine (output below). 

Does it shows any error on your machine?

```
% npm run format

> final-space-api-frontend@0.0.0 format /Users/admin/Desktop/final-space-api/frontend
> prettier --write './**/*.{js,jsx,css,md,json}'

.eslintrc.js 49ms
.prettierrc.json 4ms
babel.config.js 10ms
blog/2020-10-04-welcome.md 57ms
blog/2020-10-07-new-characters.md 14ms
blog/2020-10-07-pwa-support.md 9ms
blog/2020-10-08-episode-endpoint.md 28ms
blog/2020-10-08-images-consistent.md 12ms
blog/2020-10-10-location-endpoint-added.md 31ms
docs/about.md 28ms
docs/character.md 57ms
docs/episode.md 50ms
docs/intro.md 27ms
docs/location.md 40ms
docusaurus.config.js 31ms
package-lock.json 446ms
package.json 5ms
README.md 14ms
sidebars.js 3ms
src/css/custom.css 43ms
src/pages/Components/_styles.module.css 9ms
src/pages/Components/Body/_Body.js 11ms
src/pages/Components/Body/_Card.js 20ms
src/pages/Components/Body/_CardList.js 3ms
src/pages/Components/Body/_InfoCointainer.js 10ms
src/pages/Components/Head/_Head.js 13ms
src/pages/index.js 6ms
static/manifest.json 10ms
% 
```

Comment (lelouchB) [reply frontend/package.json:13]: Yes, 
```
$ npm run format

> final-space-api-frontend@0.0.0 format D:\code\API\final-space-api\final-space-api\frontend
> prettier --write './**/*.{js,jsx,css,md,json}'

[error] No files matching the pattern were found: "'./**/*.{js,jsx,css,md,json}'".
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! final-space-api-frontend@0.0.0 format: `prettier --write './**/*.{js,jsx,css,md,json}'`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the final-space-api-frontend@0.0.0 format script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:

```

Comment (binkpitch) [reply frontend/package.json]: I was able to reproduce the error by switching to a Windows machine (I usually code on Mac).

Can you pull this branch again and see if it works.

**Comment by lelouchB** [comment]
@binkpitch 
This looks great. 👍 

Thanks for your contribution. :)

**Pull request merged**